File: UserManagment/UserManagment.Application/Contracts/Responses/NotificationResponse.cs
namespace UserManagment.Application.Contracts.Responses;

public class NotificationResponse
{
    public Guid Id { get; set; }
    public string Type { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Message { get; set; } = string.Empty;
    public bool IsRead { get; set; }
    public Guid? RelatedEntityId { get; set; }
    public DateTime CreatedAtUtc { get; set; }
    public DateTime? ReadAtUtc { get; set; }
    public string? ActionPath { get; set; }
}

File: notification-service/src/main/java/com/mentorhub/notificationservice/dto/NotificationResponse.java
package com.mentorhub.notificationservice.dto;

import lombok.*;
import java.time.Instant;
import java.util.UUID;

@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class NotificationResponse {
    private UUID id;
    private String type;
    private String title;
    private String message;
    private boolean isRead;
    private UUID relatedEntityId;
    private Instant createdAtUtc;
    private Instant readAtUtc;
    private String actionPath;
}
